src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
> What is the specific problem with this line as-is? Perhaps we should stay with it until there's a proven problem in the wild that requires a stricter and more complex check.

+ 1

This is basically what I was getting at above: you'll never know until you try!

> Many people still support IE 8

jQuery 3 requires the jQuery 3 Compat shim to support IE8. Surely that shim can do some extra protection work if necessary?